### PR TITLE
Sort specs by line to preserve ordering

### LINKED ISSUE #519: specs should be displayed in the same order as they have been defined
See the example of Elixir.Range.range?/1

https://github.com/elixir-lang/elixir/blob/b6dc389f5e3eb2286885c172009333ef414d7860/lib/elixir/lib/range.ex#L73-L74
```elixir
@spec range?(%Range{}) :: true
@spec range?(term) :: false
```

and how it's showed in the docs
http://elixir-lang.org/docs/master/elixir/Range.html#range?/1
```
Specs:
range?(term) :: false
range?(%Range{first: term, last: term}) :: true
```

the order changes the meaning.

### PR DESCRIPTION
**Pull request by bjfish**
Fixes #519

### CODE REVIEW

File: lib/ex_doc/retriever.ex [hunk 1]
```diff
@@ -193,6 +193,7 @@ defmodule ExDoc.Retriever do
     specs = all_specs
             |> Map.get(function, [])
             |> Enum.map(&Typespec.spec_to_ast(name, &1))
+            |> Enum.sort( fn({_,l,_}, {_,l2,_}) -> l[:line] < l2[:line] end )
```
Comment (josevalim) [inline lib/ex_doc/retriever.ex]: I wonder if they are just in reverse order and calling `Enum.reverse` would be enough?

**Comment by bjfish** [comment]
I've updated to use `Enum.reverse`. The test still passes.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by eksperimental** [comment]
Thank you @bjfish

**Pull request merged**